fish-rust/src/builtins/cd.rs

```diff
@@ -9,10 +9,11 @@ use crate::{
 };
 use errno::{self, Errno};
 use libc::{fchdir, EACCES, ELOOP, ENOENT, ENOTDIR, EPERM, O_RDONLY};
+use std::rc::Rc;
 
 // The cd builtin. Changes the current directory to the one specified or to $HOME if none is
 // specified. The directory can be relative to any directory in the CDPATH variable.
-pub fn cd(parser: &mut parser_t, streams: &mut io_streams_t, args: &mut [&wstr]) -> Option<c_int> {
+pub fn cd(parser: &Parser, streams: &mut IoStreams, args: &mut [&wstr]) -> Option<c_int> {
```

Comment:
I don't remember, would need to try. Maybe because it allowed a more direct port of the operation context, which is passed by const reference. C++ allows mutating the operation context's parser field because it's inside a shared pointer but Rust doesn't. We'd add a Mutex to `ParserRef`.